### PR TITLE
Change the usage of Random class to RNGCryptoServiceProvider

### PR DESCRIPTION
**Pull request by cheeyixu**
First I tried naive way of directly setting BigInteger.data via new method setData. After that I overloaded genRandomBits in BigInteger class and found it much more simpler than the previous approach. I think the second approach ensure more encapsulation in protecting BigInteger.data being modified by external classes.

### CODE REVIEW

File: PaillierExt/PaillierManaged.cs [hunk 1]
```diff
@@ -54,43 +54,46 @@ public override string KeyExchangeAlgorithm
         }
 
         // TODO: check again for Miu
-        // p_key_strength in normal case is passed in by keysizevalue, which is 1024
+        // p_key_strength in normal case is passed in by keysizevalue, which is 102
```
Comment (bazzilic) [inline PaillierExt/PaillierManaged.cs]: typo?

**Comment by bazzilic** [comment]
I think there's a typo in your PR. Apart from that, LGTM. Will pull after I pull https://github.com/bazzilic/BigInteger/pull/18

**Comment by cheeyixu** [comment]
yes that's a typo

**Comment by bazzilic** [comment]
Resolves #4

**Pull request merged**